gix-config/src/traits.rs

```diff
@@ -0,0 +1,11 @@
+use bstr::BStr;
+
+/// Parsed elements of a Git config key, like `remote.origin.url` or `core.bare`.
+pub trait Key<'a> {
+    /// The name of the section key, like `url` in `remote.origin.url`.
+    fn name(&self) -> &'a str;
```

Comment:
Is `name()` to generic given that we implement it on `&[b]str`? `.key_name()`  might be better?